### PR TITLE
PT-153251639 Use target from previous block in target validation

### PR DESCRIPTION
**Pull request by hanssv**
https://www.pivotaltracker.com/story/show/153251639

### CODE REVIEW

File: apps/aecore/src/aec_target.erl [hunk 1]
```diff
@@ -35,10 +35,15 @@ recalculate(Header, InitialHeader) ->
...
-verify(Header, InitialHeader) ->
+-spec verify(header(), header(), header()) -> ok | {error, target_too_high}.
+verify(Header, PrevHeader, InitialHeader) ->
     HeaderTarget = aec_headers:target(Header),
-    ExpectedTarget = recalculate(Header, InitialHeader),
+    PrevTarget   = aec_headers:target(PrevHeader),
+
+    %% Block candidate target was computed using the target of the previous block
+    %% we have to do the same thing here.
+    ExpectedTarget = recalculate(Header#header{ target = PrevTarget }, InitialHeader),
```
Comment (happi) [inline apps/aecore/src/aec_target.erl]: Ì am slightly uneasy about the use of the header record directly here.
Perhaps create a set_target function in aec_header.erl

**Comment by lucafavatella** [comment]
> This is not correct. The delta header is calculated from the header when it should be calculated from the prev header if I understand the situation right.

In mining, [the delta header is computed from the block with the height already incremented](https://github.com/aeternity/epoch/blob/420973b7386c77aee29e9334bd7e00f193b87bed/apps/aecore/src/aec_mining.erl#L104), so this looks correct to me.

**Comment by lucafavatella** [comment]
@hanssv Does this PR need to bump the chain DB folder? (reviewing other PR right now, did not double check)

File: apps/aecore/src/aec_target.erl [hunk 2]
```diff
@@ -35,10 +35,16 @@ recalculate(Header, InitialHeader) ->
...
-verify(Header, InitialHeader) ->
+-spec verify(header(), header(), header()) -> ok | {error, target_too_high}.
+verify(Header, PrevHeader, InitialHeader) ->
     HeaderTarget = aec_headers:target(Header),
-    ExpectedTarget = recalculate(Header, InitialHeader),
+    PrevTarget   = aec_headers:target(PrevHeader),
+
+    %% Block candidate target was computed using the target of the previous block
+    %% we have to do the same thing here.
+    ExpectedTarget = recalculate(aec_headers:set_header(Header, PrevTarget),
```
Comment (lucafavatella) [inline apps/aecore/src/aec_target.erl]: set_target?

**Comment by hanssv** [comment]
Fixed typo and bumped the chain

**Comment by lucafavatella** [comment]
I killed the 4 OSX jobs of this PR as not needed - PR needs to be updated with test fix anyway.

**Pull request merged**